receiver/awsxrayreceiver/internal/udppoller/poller.go

```diff
@@ -173,6 +172,7 @@ func (p *poller) poll() {
 				p.receiverInstanceName,
 				Transport,
 				obsreport.WithLongLivedCtx())
+			buffer := make([]byte, pollerBufferSizeKB)
```

Comment:
Thanks - didn't realize we have concurrent pollers. Instead of making the buffer every single time, can you move buffer creation to line 121 where the polling routines are initialized and pass in as a parameter to `poll()`?

Reply:
Actually - nevermind, it shouldn't matter. Can you point me at the race condition? This loop should not be called concurrently so there's something else going on.

Reply:
I guess it's because we pass the buffer to here

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/65f181da1a66b54a83da7c1d06fec69bf8651403/receiver/awsxrayreceiver/receiver.go#L129

How about doing the translation inline without the jump? There doesn't seem to be that much point in having multiple pollers anyways if we end up just translating in a single loop.

Reply:
Yes, you are right, we actually pass the body of the buffer to translator. And because they are different threads which caused the race condition.  I didn't design the receiver, so I don't know why we have multiple pollers but just one translator. There may have some special purpose to design it in that way. However we just released it, and Min @mxiamxia believes if we don't fix it as soon as possible, we should revert the release, so I think we can fix it in this way for now.

Reply:
The performance penalty for reallocating every iteration seems extremely high though. I guess it's better than broken but I wouldn't release the version with this code either. Can you commit to optimizing this right away?

Reply:
Sure, I will optimize this.